### PR TITLE
result type and add methods

### PR DESCRIPTION
**Pull request by gragusa**
@spencerlyon2 I have added the return type and added vcov. It needs testing (especially the vcov methods).

### CODE REVIEW

**Comment by sglyon** [comment]
Excellent, I'll take a look right now.

File: src/GMM.jl [hunk 1]
```diff
@@ -83,43 +103,67 @@ function gmm(mf::Function, theta::Vector, theta_l::Vector, theta_u::Vector,
...
+                                    nmom, npar, nobs)
+    GMMEstimator(NLPE, r)
+end              
+
+status(me::MomentEstimator) = me.r.status
+StatsBase.coef(me::MomentEstimator) = me.r.coef
+objval(me::MomentEstimator) = me.r.objval
+momentfunction(me::MomentEstimator) = me.e.mf(coef(me))
+jacobian(me::MomentEstimator) = me.e.Dmf(coef(me))
+mfvcov(me::MomentEstimator, k::RobustVariance) = vcov(momentfunction(me), k)
```
Comment (sglyon) [inline src/GMM.jl:147]: `CovarianceMatrices` doesn't seem to export `RobustVariance`. We could either change that package so it does or define `const RobustVariance = CovarianceMatrices.RobustVariance` at the top of this file. What do you think it best?

**Comment by sglyon** [comment]
I'm a bit concerned about the `vcov` method for the moment estimator.

I'm trying to replicate some results I obtain using the `gmm` R package, and I am not getting the same standard errors they have for my test case. What formula are you using for the form of the covariance matrix?

File: src/GMM.jl [hunk 2]
```diff
@@ -83,43 +103,67 @@ function gmm(mf::Function, theta::Vector, theta_l::Vector, theta_u::Vector,
...
+nobs(me::MomentEstimator) = me.r.nobs
+npar(me::MomentEstimator) = me.r.npar
+
+function StatsBase.vcov(me::MomentEstimator, k::RobustVariance)
+    G = jacobian(me)
+    S =  PDMat(mfvcov(me, k))
```
Comment (sglyon) [inline src/GMM.jl:153]: I am testing a model against the `gmm` package from R. 

As it currently sits, the vcov matrix is very different than what R reports. However, if I change this line to `S =  PDMat(mfvcov(me, k)) * nobs(me)` I get very close to what R gets (close enough to call it a difference in the algorithm/kernel).

Can you see any reason why multiplying by nobs  would be correct here?

File: src/GMM.jl (hunk 1 above)
Comment (gragusa) [inline src/GMM.jl:147]: Yes. Need to invoke `CovarainceMatrices.RobustVariance` is the supertype fro all the variances defined in `CovarianceMatrices`

File: src/GMM.jl (hunk 2 above)
Comment (gragusa) [inline src/GMM.jl:153]: No fully tested. I will fix it.

**Pull request merged**